arkouda/categorical.py

```diff
@@ -626,6 +628,50 @@ def is_registered(self) -> np.bool_:
 
         return np.bool_(np.any(parts_registered))
 
+    def _list_component_names(self) -> List[str]:
+        """
+        Internal Function that returns a list of all component names
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        List[str]
+            List of all component names
+        """
+        return list(itertools.chain.from_iterable([getattr(self, p)._list_component_names() for p in Categorical.RegisterablePieces]))
```

Comment:
I had to remind myself what itertools.chain was doing here, it flattens out the list of lists and returns the components / nested-subcomponents like this:
```
['C.permutation', 'C.categories.offsets', 'C.categories.bytes', 'C.segments', 'C.codes']
```
Nice.